crypto/hash/sha256.go

```diff
@@ -84,9 +84,9 @@ func (h SHA256Hash) Compare(other SHA256Hash) int {
 
 // Xor returns the xor of this Hash and all others. It does not change this Hash.
 func (h SHA256Hash) Xor(others ...SHA256Hash) SHA256Hash {
-	for _, o := range others {
-		for i := range o {
-			h[i] ^= o[i]
+	for n := range others {
+		for i := range others[n] {
+			h[i] ^= others[n][i]
```

Comment:
Can highly optimise this with 64-bit operations. Compiler wil even vectorise if done right.

```go
o := binary.LittleEndian
o.PutUint64(h[:8], o.Uint64(h[:8]) ^ o.Uint64(other[:8]))
o.PutUint64(h[8:16], o.Uint64(h[8:16]) ^ o.Uint64(other[8:16]))
…
```